server/service/microsoft_mdm.go

```diff
@@ -1181,17 +1187,238 @@ func (svc *Service) GetMDMWindowsTOSContent(ctx context.Context, redirectUri str
+	}
+
+	err := msg.IsValidBody()
+	if err != nil {
+		return fmt.Errorf("syncml msg body is invalid %w", err)
+	}
+
+	// Get cmds to be tracked
+	protoCMDs := msg.SyncBody.Raw
+
+	if len(protoCMDs) == 0 {
+		// no cmds to track
+		return nil
+	}
+
+	// Get the DeviceID
+	deviceID, err := msg.GetSource()
+	if err != nil || deviceID == "" {
+		return fmt.Errorf("invalid SyncML message %w", err)
+	}
+
+	// Get SessionID
+	sessionID, err := msg.GetSessionID()
+	if err != nil {
+		return fmt.Errorf("session ID processing error %w", err)
+	}
+
+	// Get MessageID
+	messageID, err := msg.GetMessageID()
+	if err != nil {
+		return fmt.Errorf("message ID processing error %w", err)
+	}
+
+	// Iterate over the operations and store on DB the ones that need to be tracked
+	for _, protoCMD := range protoCMDs {
+		cmdVerb := protoCMD.XMLName.Local
+		if protoCMD.ShouldBeTracked(cmdVerb) {
+
+			// Save the response SyncML protocol commands to the DB
+			// This will help us to track the message status
+
+			newCmd := &fleet.MDMWindowsCommand{
+				CommandUUID:  protoCMD.UUID,
+				DeviceID:     deviceID,
+				SessionID:    sessionID,
+				MessageID:    messageID,
+				CommandID:    protoCMD.CmdID,
+				CmdVerb:      cmdVerb,
+				SettingURI:   protoCMD.GetTargetURI(),
+				SettingValue: protoCMD.GetTargetData(),
+				SystemOrigin: protoCMD.SystemOrigin,
+			}
+
+			err := svc.ds.MDMWindowsInsertCommand(ctx, newCmd)
+			if err != nil {
+				return fmt.Errorf("there was an issue inserting the command %v", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// getManagementResponse returns a valid SyncML response message
+func (svc *Service) getManagementResponse(ctx context.Context, reqMsg *fleet.SyncML) (*mdm_types.SyncML, error) {
+	if reqMsg == nil {
 		return nil, fleet.NewInvalidArgumentError("syncml req message", "message is not present")
 	}
 
-	// This is a placeholder for the management response message logic
+	// Get the DeviceID
+	deviceID, err := reqMsg.GetSource()
+	if err != nil || deviceID == "" {
+		return nil, fmt.Errorf("invalid SyncML message %w", err)
+	}
+
+	// Process the incoming MDM protocol commands and get the response MDM protocol commands
+	resIncomingCmds, err := svc.processIncomingMDMCmds(deviceID, reqMsg)
+	if err != nil {
+		return nil, fmt.Errorf("message processing error %w", err)
+	}
+
+	// Process the pending operations and get the MDM response protocol commands
+	resPendingCmds, err := svc.getPendingMDMCmds(ctx, deviceID)
+	if err != nil {
+		return nil, fmt.Errorf("message processing error %w", err)
+	}
+
+	// Combined cmd responses
+	resCmds := append(resIncomingCmds, resPendingCmds...)
 
-	// Raw Response
-	responseRaw := ""
+	// Create the response SyncML message
+	msg, err := svc.createResponseSyncML(ctx, reqMsg, resCmds)
+	if err != nil {
+		return nil, fmt.Errorf("message syncML creation error %w", err)
+	}
+
+	// Track the response SyncML protocol commands
+	err = svc.storeSyncMLCmd(ctx, msg)
```

Comment:
So we'll still have to account for any issues sending those commands back to the host, once they've been stored in the non-pending table, right? I think it's fine to address this later as we implement and learn more about the whole windows mdm thing, but something to keep in mind, maybe just leaving a TODO comment for that (e.g. maybe a cron that will look at `windows_mdm_commands` without results for a long time, or we add those without results to the list of pending ones when the host checks-in without sending a response for those commands?).

Reply:
> So we'll still have to account for any issues sending those commands back to the host, once they've been stored in the non-pending table, right?

Yes, once sent, the command are stored on the non-pending table with protocol information (session id, message id and cmd id) but with no return error code. The return code (`rx_error_code` on the DB) is null until a response for a given command is retrieved.

> maybe just leaving a TODO comment for that

Added

> maybe a cron that will look at windows_mdm_commands without results for a long time, or we add those without results to the list of pending ones when the host checks-in without sending a response for those commands?).

Yes, this is what the retry logic should do. The retry logic could use the session ID information to determine if retried should be performed (response for a given command should happen within a given session)